kiali/kiali
# Add proxy status in the health

Fixes https://github.com/kiali/kiali/issues/1615
Needs https://github.com/kiali/kiali-ui/pull/1884
Needs https://github.com/kiali/kiali-operator/pull/140
Needs https://github.com/kiali/helm-charts/pull/9

Steps to help testing this PR (bottom of the description) :arrow_down: 

This PR incorporates the proxy status to the health. 

![Screenshot of Kiali Console (52)](https://user-images.githubusercontent.com/613814/94037164-5e422800-fdc5-11ea-9768-3909f2ca6d9c.png)
![Screenshot of Kiali Console (51)](https://user-images.githubusercontent.com/613814/94037166-5edabe80-fdc5-11ea-99de-5642c0b1d4f5.png)
![Screenshot of Kiali Console (50)](https://user-images.githubusercontent.com/613814/94037167-5f735500-fdc5-11ea-8b74-e970168fb655.png)
![Screenshot of Kiali Console (49)](https://user-images.githubusercontent.com/613814/94037170-5f735500-fdc5-11ea-86b4-5a53316b68d1.png)
![Screenshot of Kiali Console (48)](https://user-images.githubusercontent.com/613814/94037171-5f735500-fdc5-11ea-80c6-15e91e84877c.png)

Steps to Test:
My approach to test this is creating a sidecar object on one workload and update it many times and in a short time period.
I created the following script to force productpage to be in `STALE` status: https://github.com/xeviknal/istio-lab/tree/master/kiali-1615

1. Download both files in the link.
2. Run the script.
3. If you want to start the test again, update the bind-sidecar.yaml to have `bookinfo/reviews-v0` as a host. Go to step 2.

The status shown in Kiali should match the resulting status in the following command: `istioctl proxy-status`. Take in account that the is a cache that might delay some changes.


TODO:
- [x] Add test to the current work
- [x] Proxy status at Application level (list and details page)
- [x] Proxy status at Overview page (app, workload level)
- [x] Edit permissions to allow the pod/proxy subresource (https://github.com/kiali/kiali-operator/pull/140)

Subsequent PR:
- see the [diff](https://istio.io/latest/docs/ops/diagnostic-tools/proxy-cmd/#retrieve-diffs-between-envoy-and-istiod) / [related](https://github.com/kiali/kiali/issues/3203)
